src/esp/sim/Simulator.cpp

```diff
@@ -583,6 +592,140 @@ void Simulator::reconfigureReplayManager(bool enableGfxReplaySave) {
       });
 }
 
+void Simulator::updateShadowMapDrawableGroup() {
+  scene::SceneGraph& sg = getActiveSceneGraph();
+  // currently the method is naive: destroy the existing group, and recreate one
+  sg.deleteDrawableGroup(shadowMapDrawableGroupName);
+  sg.createDrawableGroup(shadowMapDrawableGroupName);
+  const gfx::DrawableGroup& sourceGroup = sg.getDrawables();
+  gfx::DrawableGroup* shadowMapGroup =
+      sg.getDrawableGroup(shadowMapDrawableGroupName);
+  CORRADE_INTERNAL_ASSERT(shadowMapGroup);
+
+  for (size_t iDrawable = 0; iDrawable < sourceGroup.size(); ++iDrawable) {
+    const gfx::Drawable& currentDrawable =
+        static_cast<const gfx::Drawable&>(sourceGroup[iDrawable]);
+    gfx::DrawableType type = currentDrawable.getDrawableType();
+    // So far no support for the PTex (lighting-baked mesh)
+    // SKIP!!!
+    if ((type != gfx::DrawableType::Generic) &&
+        (type != gfx::DrawableType::Pbr)) {
+      continue;
+    }
+
+    esp::scene::SceneNode& node = currentDrawable.getSceneNode();
+    node.addFeature<gfx::VarianceShadowMapDrawable>(
+        &currentDrawable.getMesh(), resourceManager_->getShaderManager(),
+        shadowMapGroup);
+  }
+}
+
+void Simulator::computeShadowMaps(float lightNearPlane, float lightFarPlane) {
+  scene::SceneGraph& sg = getActiveSceneGraph();
+  auto& shadowManager = resourceManager_->getShadowMapManger();
+  auto& shadowMapKeys = resourceManager_->getShadowMapKeys();
+  std::vector<Mn::ResourceKey>& keys = shadowMapKeys[activeSceneID_];
+
+  // XXXXXXXXXXXX
+  // I need John's help. How to get the light positions here??!!
+  // currently the light position is hard coded for a particular replicaCAD
+  // scene
+  Mn::Vector3 lightPos[3] = {Mn::Vector3{2.77, 2.77, 5.16},
```

Comment:
@jturner65 : John, I need your help here. 

I am going to compute the shadow maps for the lights (1 map per light), and I would like to obtain the light absolute position specified in the `scene_dataset` config.

As you can see here, I hard-coded 3 light positions here to do the demo. But this is definitely wrong for a new scene.
I need a generic method to query this value (light position) within the Simulator class.

As you can see, currently this function `computeShadowMaps` is called in the viewer after the `Simulator` instance is created. So I assume `scene_dataset` config has been parsed and the light positions are already stored somewhere in some data structure. I just need a way to get the data.
Let me know how I can do it. Thanks!

Reply:
Ok, to get the LightSetup that is currently in use you can use the same mechanism by which it is created (which will is performed when Scene is instantiated).

In c++ Simulator class, config_.sceneLightSetup will always hold the key of the current scene's LightSetup
Use the following function to acquire a copy of the LightSetup for the current scene, built from the attributes  : 
```
 esp::gfx::LightSetup lightingSetup =
          metadataMediator_->getLightLayoutAttributesManager()
              ->createLightSetupFromAttributes(config_.sceneLightSetup);
```